src/corecel/Macros.hh

```diff
@@ -214,3 +214,27 @@
             HANDLE_EXCEPTION(std::current_exception()); \
         }                                               \
     } while (0)
+
+/*!
+ * \def CELER_TRY_HANDLE_CONTEXT
+ *
+ * Try the given statement, and if it fails, chain it into the given exception.
+ *
+ * The given \c CONTEXT_EXCEPTION must be an expression that yields an rvalue
+ * to a \c std::exception subclass that isn't \c final . The resulting chained
+ * exception will be passed into \c HANDLE_EXCEPTION for processing.
+ */
+#define CELER_TRY_HANDLE_CONTEXT(                         \
```

Comment:
```suggestion
#define CELER_TRY_RETHROW_WITH_CONTEXT(                         \
```
(That is too long but)  `HANDLE` implies that the exception is treated and resolved rather than augmented and rethrown.   i.e with the current name I would assume that the code never throw/leak any exception whereas the actual behavior is the same as without the try-catch, just with added information. (So another name optin `CELER_TRY_ADD_CONTEXT`

Reply:
No, this does actually handle and should not rethrow. It's really more like a `CELER_TRY_RETHROW_WITH_CONTEXT` inside of a `CELER_TRY_HANDLE_CONTEXT`.

Reply:
> should not rethrow.

humm ... I am confused ... `std::throw_with_nested(CONTEXT_EXCEPTION);` is literally throw (again) with additional information.

> HANDLE any exceptions with a function

humm .. I am seemingly mis-reading the code ... I don't se how `HANDLE_EXCEPTION` can be reached if `STATEMENT` throw an exception ... but all its exception are caught by `catch(...)` and then thrown via the `throw_with_nested`  ... what am I missing?

Reply:
ah .. sigh .. nevermind, I missed the `CELER_TRY_HANDLE` part of the macro.